### ISSUE TITLE
Keycloak Auth and Post / Delete / Put / Patch requests

### ISSUE THREAD

**Issue by ClementineBiais** [TypedProject/ts-express-decorators#1824]
Good afternoon,

### Information
- Version:: 6.95.x
- Packages: platform-express

While implementing a Keycloak RESTful API, a 404 error occurs on every route but HEAD, OPTIONS and GET ones.

### Examples

I am trying to list entities with a get /entities

`
  @Get()
  @KeycloakAuth({ role: "entities.read" })
  async getEntityList(...) { ... }
`

I get a 200 response. 

I then try to add a new entity to that list with the same permission.
`
  @Post()
  @KeycloakAuth({ role: "entities.read" })
  async createEntity() { 
     // empty method to make sure it does not come from the content
  }
`

I get a 404 answer :

`{
  "name": "NOT_FOUND",
  "message": "Resource \"/entities\" not found",
  "status": 404,
  "errors": [],
  "stack": "NOT_FOUND: Resource \"/entities\" not found\n    at PlatformExceptions.resourceNotFound (/home/cbiais/Documents/mazedia/workspace/git/user-management-api/node_modules/@tsed/platform-exceptions/src/services/PlatformExceptions.ts:51:23)\n    at /home/cbiais/Documents/mazedia/workspace/git/user-management-api/node_modules/@tsed/platform-express/src/components/PlatformExpress.ts:100:81\n    at Layer.handle [as handle_request] (/home/cbiais/Documents/mazedia/workspace/git/user-management-api/node_modules/express/lib/router/layer.js:95:5)\n    at trim_prefix (/home/cbiais/Documents/mazedia/workspace/git/user-management-api/node_modules/express/lib/router/index.js:323:13)\n    at /home/cbiais/Documents/mazedia/workspace/git/user-management-api/node_modules/express/lib/router/index.js:284:7\n    at Function.process_params (/home/cbiais/Documents/mazedia/workspace/git/user-management-api/node_modules/express/lib/router/index.js:341:12)\n    at next (/home/cbiais/Documents/mazedia/workspace/git/user-management-api/node_modules/express/lib/router/index.js:275:10)\n    at /home/cbiais/Documents/mazedia/workspace/git/user-management-api/node_modules/express/lib/router/index.js:641:15\n    at next (/home/cbiais/Documents/mazedia/workspace/git/user-management-api/node_modules/express/lib/router/index.js:260:14)\n    at protect (/home/cbiais/Documents/mazedia/workspace/git/user-management-api/node_modules/keycloak-connect/middleware/protect.js:54:16)\n    at PlatformExpressHandler.callReturnedMiddleware (/home/cbiais/Documents/mazedia/workspace/git/user-management-api/node_modules/@tsed/common/src/services/PlatformHandler.ts:228:12)\n    at PlatformExpressHandler.onSuccess (/home/cbiais/Documents/mazedia/workspace/git/user-management-api/node_modules/@tsed/common/src/services/PlatformHandler.ts:210:19)\n    at /home/cbiais/Documents/mazedia/workspace/git/user-management-api/node_modules/@tsed/common/src/services/PlatformHandler.ts:156:29\n    at processTicksAndRejections (node:internal/process/task_queues:96:5)"
}`

I couldn't find any way to fix it.

**Comment by Romakita** [comment]
@xCryzed have an idea about this problem?

**Comment by xCryzed** [comment]
@ClementineBiais I think the 404 occurs because your method body is empty.

Does a client named `entities` in your running keycloak instance exist? You also need the corresponding client role `read`.

**Comment by ClementineBiais** [comment]
@xCryzed thank you for your answer.
I emptied my method body because I wanted to make sure that I wasn't related. But the same error occurs when it isn't empty, which is why I don't think it is related.
Elsewhere, no client named `entities` exists in my Keycloak instance, because I have another client name that I sliced here (the real format is `client:entities.read,` and I am sure I have a corresponding role because it works with the get route.
